extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/KinesisRecordSupplier.java

```diff
@@ -803,11 +805,21 @@ private Set<Shard> getShardsUsingListShards(String stream)
   public Set<String> getPartitionIds(String stream)
   {
     return wrapExceptions(() -> {
-      ImmutableSet.Builder<String> partitionIds = ImmutableSet.builder();
-      for (Shard shard : getShards(stream)) {
-        partitionIds.add(shard.getShardId());
+      final Set<String> retVal = new TreeSet<>();
+      ListShardsRequest request = new ListShardsRequest().withStreamName(stream);
+      while (true) {
+        ListShardsResult result = kinesis.listShards(request);
+        retVal.addAll(result.getShards()
+                            .stream()
+                            .map(Shard::getShardId)
+                            .collect(Collectors.toList())
+        );
+        String nextToken = result.getNextToken();
+        if (nextToken == null) {
+          return retVal;
+        }
+        request = new ListShardsRequest().withNextToken(nextToken);
```

Comment:
what is the difference between this and `getShardsUsingListShards`?

Reply:
Thanks for pointing it out. This is an error and would always use listShards to get partition Ids